Public/Src/Engine/Processes/FileAccessManifest.cs

```diff
@@ -281,21 +459,47 @@ private static void WriteChars(BinaryWriter writer, string str)
             }
         }
 
+        private static string ReadChars(BinaryReader reader)

```

Comment:
>ReadChars [](start = 30, length = 9)

This is so generic, shouldn't these be extension methods or use our BuildXLReader/Writer with their helpers?

Reply:
BinaryWriter/Reader have already methods for writing/reading string.

What I'm doing here is reuse WriteChars method that is used to serialize manifest to Detours code. By reusing, I introduced ReadChars that "matches" WriteChars. Yes, I'm aware of the asymmetry, and will add comment about that. But, changing WriteChars entails changing the code in Detours as well. I'm not doing that right now.

---
In reply to: [276812324](https://github.com/Microsoft/BuildXL/pull/159#discussion_r276812324) [](ancestors = 276812324)